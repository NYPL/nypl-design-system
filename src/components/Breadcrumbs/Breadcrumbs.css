@import "../Icons/Icon.css";
@import "../../styles/base/_02-breakpoints.css";
@import "../../styles/base/_03-mixins.css";

.breadcrumbs {
  @add-mixin breakout;

  background-color: var(--ui-black);
  padding-bottom: var(--space-xs);
  padding-top: var(--space-xs);

  &__item {
    align-items: center;
    display: inline;
    font-size: var(--font-size--1);
    line-height: 1.5;
    margin-right: var(--space-xxs);
    position: relative;
    word-break: break-word;

    &:not(:last-child) {
      &::after {
        content: "/";
        padding-left: var(--space-xxs);
      }

      @media (--medium-break) {
        display: none;
      }
    }

    &:last-child {
      @media (--medium-break) {
        padding-left: var(--space-m);
      }
    }
  }

  &__icon {
    @add-mixin icon;

    left: 0;
    position: absolute;
    top: 50%;
    transform: rotate(90deg) translateX(-50%);

    @media (--medium-break) {
      display: none;
    }
  }

  &__link,
  &__link:link,
  &__link:visited,
  &__link:focus {
    color: inherit;
    text-decoration: none;
  }

  &__link:hover {
    text-decoration: underline;
  }

  &__list {
    @add-mixin list-reset;
    @add-mixin wrapper;

    color: var(--ui-white);
  }

  .nypl--books-and-more & {
    background-color: var(--section-books-and-more-secondary);
  }

  .nypl--locations & {
    background-color: var(--section-locations-secondary);
  }

  .nypl--research & {
    background-color: var(--section-research-secondary);
  }

  .nypl--whats-on & {
    background-color: var(--section-whats-on-secondary);
  }
}

.storybook-breadcrumbsExample {
  .breadcrumbs {
    flex: 1 1 50%;
    margin-left: unset;
    margin-right: var(--space-l);
    max-height: 42px;
  }
}
