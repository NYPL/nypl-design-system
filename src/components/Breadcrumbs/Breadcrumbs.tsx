import * as React from "react";
import bem from "../../utils/bem";
import Icon from "../Icons/Icon";
import { IconNames } from "../Icons/IconTypes";

type breadcrumb = { url: string; text: string };

function isTextBreadcrumb(obj: breadcrumb | JSX.Element): obj is breadcrumb {
  return typeof (obj as breadcrumb).url === "string";
}

export interface BreadcrumbProps {
  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
  blockName?: string;
  /** Breadcrumb links as an array */
  breadcrumbs: breadcrumb[] | JSX.Element[];
  /** className you can add in addition to 'input' */
  className?: string;
  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
  modifiers?: string[];
}

export default class Breadcrumbs extends React.Component<BreadcrumbProps, any> {
  static defaultProps = {};

  constructor(props: BreadcrumbProps) {
    super(props);
  }

  render(): JSX.Element {
    const { breadcrumbs, className, blockName, modifiers } = this.props;

    const baseClass = "breadcrumbs";

    const breadcrumbItems = [];

    if (!breadcrumbs || breadcrumbs.length === 0) {
      throw new Error(
        "Breadcrumbs must contain a set of links. Breadcrumbs currently empty"
      );
    }
    breadcrumbs.forEach((item: breadcrumb | JSX.Element, index: number) => {
      const last = index === breadcrumbs.length - 1;
      let linkComponent: string | JSX.Element;

      if (isTextBreadcrumb(item)) {
        linkComponent = item.url ? (
          <a href={item.url} className={bem("link", [], baseClass)}>
            {item.text}
          </a>
        ) : (
          item.text
        );
      } else {
        const props = {
          className: bem("link", [], baseClass),
        };
        linkComponent = React.createElement(
          item.type,
          { ...props, ...item.props },
          item.props.children
        );
      }
      breadcrumbItems.push(
        <li
          key={`${baseClass}-${index}`}
          className={bem("item", [], baseClass)}
        >
          {last && (
            <Icon
              name={IconNames.arrow}
              blockName={baseClass}
              modifiers={["small"]}
              decorative={true}
            />
          )}
          {linkComponent}
        </li>
      );
    });

    return (
      <nav
        className={bem(baseClass, modifiers, blockName, [className])}
        role="navigation"
        aria-label="Breadcrumbs"
      >
        <ul className={bem("list", [], baseClass)}>{breadcrumbItems}</ul>
      </nav>
    );
  }
}
