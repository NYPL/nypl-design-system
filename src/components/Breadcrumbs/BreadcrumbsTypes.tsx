export enum BreadcrumbsTypes {
  Blogs = "blogs",
  BooksAndMore = "booksAndMore",
  Education = "education",
  Locations = "locations",
  Research = "research",
  WhatsOn = "whatsOn",
}
