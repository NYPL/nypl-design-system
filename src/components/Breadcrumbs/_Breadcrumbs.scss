$break-breadcrumbs-mobile: max-width 599px;

.breadcrumbs {
  background-color: var(--ui-black);
  padding-bottom: var(--space-xs);
  padding-top: var(--space-xs);

  &__item {
    @include space-inline-xxs;

    align-items: center;
    display: inline;
    font-size: var(--font-size--1);
    line-height: 1.5;
    position: relative;
    word-break: break-word;

    &:not(:last-child) {
      &::after {
        content: "/";
        padding-left: var(--space-xxs);
      }

      @include breakpoint($break-breadcrumbs-mobile) {
        display: none;
      }
    }

    &:last-child {
      @include breakpoint($break-breadcrumbs-mobile) {
        display: block;
        padding-left: var(--space-m);
      }
    }
  }

  &__icon {
    @include icon;

    left: 0;
    position: absolute;
    top: 50%;
    transform: rotate(90deg) translateX(-50%);

    @include breakpoint($breakpoint-medium) {
      display: none;
    }
  }

  &__link,
  &__link:link,
  &__link:visited,
  &__link:focus {
    color: var(--ui-white);
    text-decoration: none;
  }

  &__item a:hover {
    color: var(--ui-gray-light);
    text-decoration: underline;
  }

  &__list {
    @include list-reset;
    @include wrapper;

    color: var(--ui-white);
  }

  .nypl--books-and-more & {
    background-color: var(--section-books-and-more-secondary);
  }

  .nypl--locations & {
    background-color: var(--section-locations-secondary);
  }

  .nypl--research & {
    background-color: var(--section-research-secondary);
  }

  .nypl--whats-on & {
    background-color: var(--section-whats-on-secondary);
  }
}

.storybook-breadcrumbsExample {
  .breadcrumbs {
    flex: 1 1 50%;
    margin-left: unset;
    margin-right: var(--space-l);
    max-height: 42px;
  }
}
