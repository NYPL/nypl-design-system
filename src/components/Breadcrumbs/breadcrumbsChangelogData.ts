/** This data is used to populate the ComponentChangelogTable component.
 *
 * date: string (when adding new entry during development, set value as "Prerelease")
 * version: string (when adding new entry during development, set value as "Prerelease")
 * type: "Bug Fix" | "New Feature" | "Update";
 * affects: array["Accessibility" | "Documentation" | "Functionality" | "Styles"];
 * notes: array (will render as a bulleted list, add one array element for each list element)
 */
import { ChangelogData } from "../../utils/ComponentChangelogTable";

export const changelogData: ChangelogData[] = [
  {
    date: "Prerelease",
    version: "Prerelease",
    type: "Update",
    affects: ["Styles"],
    notes: ["Chakra 2.8 update."],
  },
  {
    date: "2024-02-22",
    version: "2.1.6",
    type: "Update",
    affects: ["Styles"],
    notes: ["Updates focus ring color to match color of text."],
  },
];
