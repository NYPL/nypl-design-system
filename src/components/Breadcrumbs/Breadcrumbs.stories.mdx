import {
  Meta,
  Story,
  Canvas,
  ArgsTable,
  Preview,
  Description,
} from "@storybook/addon-docs/blocks";
import { withDesign } from "storybook-addon-designs";

import Breadcrumbs from "./Breadcrumbs";
import Heading from "../Heading/Heading";
import Link from "../Link/Link";
import sections from "../../utils/siteSections";

<Meta
  title="Breadcrumbs"
  component={Breadcrumbs}
  decorators={[withDesign]}
  parameters={{
    design: {
      type: "figma",
      url:
        "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Master?node-id=11895%3A547",
    },
  }}
  argTypes={{
    blockName: { table: { disable: true } },
    className: { table: { disable: true } },
    modifiers: { table: { disable: true } },
  }}
/>

# Breadcrumbs

<Description of={Breadcrumbs} />

The `Breadcrumbs` component is a navigation element that provides a breadcrumb path that reflects the site structure and allows a user to navigate to any page available in the breadcrumb hierarchy.

<Preview withToolbar>
  <Story
    name="Basic"
    args={{
      breadcrumbs: [
        { url: "#", text: "Parent" },
        { url: "#", text: "Child" },
        { url: "#", text: "Grandchild" },
      ],
    }}
  >
    {args => <Breadcrumbs {...args} />}
  </Story>
</Preview>

<ArgsTable story="Basic" />

### Long Titles

<Canvas>
  <Breadcrumbs
    breadcrumbs={[
      { url: "#", text: "Parent with a Long Name" },
      { url: "#", text: "Child with an Even Longer Name" },
      {
        url: "#",
        text: "Grandchild with an Exceptionally Long Name",
      },
      {
        url: "#",
        text:
          "Great-Grandchild with the Longest Name That Will Wrap onto the Second Line of the Breadcrumb Component Under the Right Circumstances",
      },
    ]}
  />
</Canvas>

## Color Variations

The `Breadcrumbs` component background color will match the secondary section color of the website section it is placed into. See more about brand colors <Link href="?path=/story/colors--colors-brand">here</Link>.

<Preview>
  <Story
    name="Color Variations"
    args={{
      breadcrumbs: [
        { url: "#", text: "Parent" },
        { url: "#", text: "Child" },
        {
          url: "#",
          text: "Grandchild",
        },
      ],
    }}
  >
    {args => (
      <div className="nypl--books-and-more">
        <Heading level={3}>Books and More</Heading>
        <Breadcrumbs
          breadcrumbs={[
            { url: "#", text: "Parent" },
            { url: "#", text: "Child" },
            {
              url: "#",
              text: "Grandchild",
            },
          ]}
        />
      </div>
    )}
  </Story>
</Preview>

<Canvas>
  <div className="nypl--locations">
    <Heading level={3}>Locations</Heading>
    <Breadcrumbs
      breadcrumbs={[
        { url: "#", text: "Parent" },
        { url: "#", text: "Child" },
        {
          url: "#",
          text: "Grandchild",
        },
      ]}
    />
  </div>
</Canvas>

<Canvas>
  <div className="nypl--research">
    <Heading level={3}>Research</Heading>
    <Breadcrumbs
      breadcrumbs={[
        { url: "#", text: "Parent" },
        { url: "#", text: "Child" },
        {
          url: "#",
          text: "Grandchild",
        },
      ]}
    />
  </div>
</Canvas>

<Canvas>
  <div className="nypl--whats-on">
    <Heading level={3}>What's On</Heading>
    <Breadcrumbs
      breadcrumbs={[
        { url: "#", text: "Parent" },
        { url: "#", text: "Child" },
        {
          url: "#",
          text: "Grandchild",
        },
      ]}
    />
  </div>
</Canvas>
