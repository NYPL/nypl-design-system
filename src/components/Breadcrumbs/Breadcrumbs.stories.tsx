import { Box, VStack } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";

import Breadcrumbs, { breadcrumbTypeArray } from "./Breadcrumbs";
import Heading from "../Heading/Heading";

const breadcrumbsData = [
  { url: "#", text: "Parent" },
  { url: "#", text: "Child" },
  { url: "#", text: "Grandchild" },
];
const breadcrumbsLongTextData = [
  { url: "#", text: "Parent with a Long Name" },
  { url: "#", text: "Child with an Even Longer Name" },
  {
    url: "#",
    text: "Grandchild with an Exceptionally Long Name",
  },
  {
    url: "#",
    text: "Great-Grandchild with the Longest Name That Will Wrap onto the Second Line of the Breadcrumb Component Under the Right Circumstances",
  },
];

const meta: Meta<typeof Breadcrumbs> = {
  title: "Components/Navigation/Breadcrumbs",
  component: Breadcrumbs,
  argTypes: {
    breadcrumbsData: { control: false },
    breadcrumbsType: {
      table: { defaultValue: { summary: "whatsOn" } },
      control: {
        type: "select",
      },
      options: breadcrumbTypeArray,
    },
    className: { control: false },
    id: { control: false },
  },
};

export default meta;
type Story = StoryObj<typeof Breadcrumbs>;

/**
 * Main Story for the Accordion component. This must contains the `args`
 * and `parameters` properties in this object.
 */
export const WithControls: Story = {
  args: {
    breadcrumbsData,
    breadcrumbsType: "whatsOn",
    className: undefined,
    id: "breadcrumbs-id",
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=10766%3A1031",
    },
    jest: ["Breadcrumbs.test.tsx"],
  },
};

// The following are additional Breadcrumbs example Stories.
export const LongText: Story = {
  render: () => <Breadcrumbs breadcrumbsData={breadcrumbsLongTextData} />,
};
export const ColorVariations: Story = {
  render: () => (
    <VStack align="stretch" spacing="l">
      <Box>
        <Heading level="h3" size="heading6">
          Blogs
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="blogs"
        />
      </Box>
      <Box>
        <Heading level="h3" size="heading6">
          Books and More
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="booksAndMore"
        />
      </Box>
      <Box>
        <Heading level="h3" size="heading6">
          Connect
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="connect"
        />
      </Box>
      <Box>
        <Heading level="h3" size="heading6">
          Education
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="education"
        />
      </Box>
      <Box>
        <Heading level="h3" size="heading6">
          Locations
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="locations"
        />
      </Box>
      <Box>
        <Heading level="h3" size="heading6">
          Research
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="research"
        />
      </Box>
      <Box>
        <Heading level="h3" size="heading6">
          What's On
        </Heading>
        <Breadcrumbs
          breadcrumbsData={breadcrumbsData}
          breadcrumbsType="whatsOn"
        />
      </Box>
    </VStack>
  ),
};
