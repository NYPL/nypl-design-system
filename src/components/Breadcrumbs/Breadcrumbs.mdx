import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
import ComponentChangelogTable from "../../utils/ComponentChangelogTable";

import * as BreadcrumbsStories from "./Breadcrumbs.stories";
import { changelogData } from "./breadcrumbsChangelogData";
import Link from "../Link/Link";

<Meta of={BreadcrumbsStories} />

# Breadcrumbs

| Component Version | DS Version   |
| ----------------- | ------------ |
| Added             | `0.0.3`      |
| Latest            | `Prerelease` |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#long-titles" target="_self">Long Titles</Link>}
- {<Link href="#color-variations" target="_self">Color Variations</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={BreadcrumbsStories} />

<Source
  code={`
const breadcrumbsData = [
  { url: "#", text: "Parent" },
  { url: "#", text: "Child" },
  { url: "#", text: "Grandchild" },
];
// ...
<Breadcrumbs breadcrumbsData={breadcrumbsData} />
`}
  language="tsx"
/>

## Component Props

<Canvas of={BreadcrumbsStories.WithControls} />

<ArgTypes of={BreadcrumbsStories.WithControls} />

### Accessibility

Only one `Breadcrumbs` component should be rendered on a page. This is because
only one HTML `<nav>` element with an `aria-label` attribute value of
"Breadcrumbs" should be rendered on a page. The DS `Breadcrumbs` component
renders this HTML landmark so only one component must be rendered on a page.

Internally, links are organized in a `ul` list element. The current page link is
denoted by an `aria-current` attribute value of "page".

- [W3 Breadcrumbs Practice](https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html)
- [Chakra UI Breadcrumb](https://chakra-ui.com/docs/components/navigation/breadcrumb)

## Long Titles

Ideally, breadcrumb page titles should be short and concise. In cases where a
breadcrumb page title is over 40 characters, the `Breadcrumb` component will
truncate the title and an ellipsis will be added to the end of the truncated
text. The full title will be shown in a `Tooltip` component when hovering over
the truncated page title.

<Canvas of={BreadcrumbsStories.LongText} />

## Color Variations

The `Breadcrumbs` component background color can be set to a value of the
`breadcrumbsType` type. If the `breadcrumbsType` prop is omitted, the default
background color is `"whatsOn"` (`ui.black`).

<Canvas of={BreadcrumbsStories.ColorVariations} />

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
