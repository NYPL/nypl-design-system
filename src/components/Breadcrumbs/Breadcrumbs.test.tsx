import { expect } from "chai";
import * as Enzyme from "enzyme";
import * as React from "react";

import Breadcrumbs from "./Breadcrumbs";

describe("Breadcrumbs", () => {
  let wrapper: Enzyme.ShallowWrapper<any, any>;
  const breadcrumbString = [
    { url: "#", text: "test1" },
    { url: "#", text: "test2" },
  ];
  const breadcrumbComponent = [
    <a key="link1" href="#test1">
      <span>Hello</span>
    </a>,
    <a key="link2" href="#test2">
      Goodbye
    </a>,
  ];

  it("Renders a tag with custom text", () => {
    wrapper = Enzyme.shallow(<Breadcrumbs breadcrumbs={breadcrumbString} />);
    const links = wrapper.render();

    expect(links.find("a.breadcrumbs__link")).to.have.lengthOf(2);
    expect(links.find("a.breadcrumbs__link").first().text()).to.equal("test1");
    expect(links.find("a.breadcrumbs__link").last().text()).to.equal("test2");
  });

  it("Renders breadcrumb when passed components", () => {
    wrapper = Enzyme.shallow(<Breadcrumbs breadcrumbs={breadcrumbComponent} />);
    const links = wrapper.render();
    expect(links.find("a.breadcrumbs__link").first().text()).to.equal("Hello");
  });

  it("Renders icon", () => {
    wrapper = Enzyme.shallow(<Breadcrumbs breadcrumbs={breadcrumbComponent} />);

    expect(wrapper.find("Icon")).to.have.lengthOf(1);
  });

  it("Throws error when nothing is passed into Breadcrumb", () => {
    expect(() => Enzyme.mount(<Breadcrumbs breadcrumbs={[]} />)).to.throw(
      "Breadcrumbs must contain a set of links. Breadcrumbs currently empty"
    );
  });
});
