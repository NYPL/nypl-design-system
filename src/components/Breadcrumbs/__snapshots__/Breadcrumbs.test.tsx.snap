// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Snapshot Renders the UI snapshot correctly 1`] = `
<nav
  aria-label="Breadcrumb"
  className="chakra-breadcrumb css-0"
  id="breadcrumbs-test"
>
  <ol
    className="chakra-breadcrumb__list css-0"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string1"
      >
        <span
          className="breadcrumb-label"
        >
          string1
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string2"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon breadcrumbs-icon css-onkibi"
          focusable={false}
          id="breadcrumbs-test__backarrow"
          role="img"
          title="arrow icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <span
          className="breadcrumb-label"
        >
          string2
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <span
        aria-current="page"
        className="chakra-breadcrumb__link css-0"
      >
        <span
          className="breadcrumb-label"
        >
          string3
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs Snapshot Renders the UI snapshot correctly 2`] = `
<nav
  aria-label="Breadcrumb"
  className="chakra-breadcrumb css-0"
  id="breadcrumbs-test"
>
  <ol
    className="chakra-breadcrumb__list css-0"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string1"
      >
        <span
          className="breadcrumb-label"
        >
          string1
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string2"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon breadcrumbs-icon css-onkibi"
          focusable={false}
          id="breadcrumbs-test__backarrow"
          role="img"
          title="arrow icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <span
          className="breadcrumb-label"
        >
          string2
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <span
        aria-current="page"
        className="chakra-breadcrumb__link css-0"
      >
        <span
          className="breadcrumb-label"
        >
          string3
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs Snapshot Renders the UI snapshot correctly 3`] = `
<nav
  aria-label="Breadcrumb"
  className="chakra-breadcrumb css-0"
  id="breadcrumbs-test"
>
  <ol
    className="chakra-breadcrumb__list css-0"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string1"
      >
        <span
          className="breadcrumb-label"
        >
          string1
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string2"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon breadcrumbs-icon css-onkibi"
          focusable={false}
          id="breadcrumbs-test__backarrow"
          role="img"
          title="arrow icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <span
          className="breadcrumb-label"
        >
          string2
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <span
        aria-current="page"
        className="chakra-breadcrumb__link css-0"
      >
        <span
          className="breadcrumb-label"
        >
          string3
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs Snapshot Renders the UI snapshot correctly 4`] = `
<nav
  aria-label="Breadcrumb"
  className="chakra-breadcrumb css-0"
  id="breadcrumbs-test"
>
  <ol
    className="chakra-breadcrumb__list css-0"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string1"
      >
        <span
          className="breadcrumb-label"
        >
          string1
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string2"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon breadcrumbs-icon css-onkibi"
          focusable={false}
          id="breadcrumbs-test__backarrow"
          role="img"
          title="arrow icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <span
          className="breadcrumb-label"
        >
          string2
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <span
        aria-current="page"
        className="chakra-breadcrumb__link css-0"
      >
        <span
          className="breadcrumb-label"
        >
          string3
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs Snapshot Renders the UI snapshot correctly 5`] = `
<nav
  aria-label="Breadcrumb"
  className="chakra-breadcrumb css-0"
  id="breadcrumbs-test"
>
  <ol
    className="chakra-breadcrumb__list css-0"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string1"
      >
        <span
          className="breadcrumb-label"
        >
          string1
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string2"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon breadcrumbs-icon css-onkibi"
          focusable={false}
          id="breadcrumbs-test__backarrow"
          role="img"
          title="arrow icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <span
          className="breadcrumb-label"
        >
          string2
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <span
        aria-current="page"
        className="chakra-breadcrumb__link css-0"
      >
        <span
          className="breadcrumb-label"
        >
          string3
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs Snapshot Renders the UI snapshot correctly 6`] = `
<nav
  aria-label="Breadcrumb"
  className="chakra-breadcrumb css-1f2fw9u"
  id="breadcrumbs-test"
>
  <ol
    className="chakra-breadcrumb__list css-0"
  >
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string1"
      >
        <span
          className="breadcrumb-label"
        >
          string1
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <a
        className="chakra-breadcrumb__link css-0"
        href="#string2"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon breadcrumbs-icon css-onkibi"
          focusable={false}
          id="breadcrumbs-test__backarrow"
          role="img"
          title="arrow icon"
          viewBox="0 0 24 24"
        >
          <g
            stroke="currentColor"
            strokeWidth="1.5"
          >
            <path
              d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
              fill="none"
              strokeLinecap="round"
            />
            <path
              d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
              fill="currentColor"
              strokeLinecap="round"
            />
            <circle
              cx="12"
              cy="12"
              fill="none"
              r="11.25"
              strokeMiterlimit="10"
            />
          </g>
        </svg>
        <span
          className="breadcrumb-label"
        >
          string2
        </span>
      </a>
      <span
        className="css-t4q1nq"
        role="presentation"
      >
        /
      </span>
    </li>
    <li
      className="chakra-breadcrumb__list-item css-18biwo"
    >
      <span
        aria-current="page"
        className="chakra-breadcrumb__link css-0"
      >
        <span
          className="breadcrumb-label"
        >
          string3
        </span>
      </span>
    </li>
  </ol>
</nav>
`;
